### PR TITLE
Storage: Add maxFiles to list functions

### LINKED ISSUE #71877: Canvas SVG icons picklist not showing all icons
### What happened?

Pick list for svg icons (in this case unicons) only displays a small subset of icons (from **0-plus.svg** down to **assistive-listening-systems.svg**. There are 1208 icons in the unicons directory but only the first 98 are shown. Also searching for icons that don't appear on the pick list returns nothing even though the icons exist in the unicons directory. Running in k8s but also tried 10.0.2 running locally via docker and observed same behavior.

### What did you expect to happen?

All svg icons in the unicons directory would be available to select.

### Did this work before?

Unknown, never tried canvas before 10.x

### How do we reproduce it?

1.Create canvas
2.Add icon from unicons folder
3. Observe only a small subset of icons is shown.

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

Grafana: 10.0.2
OS: Ubuntu
Browser: All

### Grafana platform?

Kubernetes

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by drew08t**
For canvas and geomap panels, icon list picker is limited to the first 100 icons. After changing maxDataPoints in the query from the frontend, it was determined that this was not being used in the backend for listing files.

This PR adds argument in file storage list functions in the backend to handle max file count based on maxDataPoints from frontend list query requests.

- [x] Add maxFiles prop to ResourceDimensionEditor to be used by canvas and geomap

BEFORE:
Oct-12-2023 13-15-26
Oct-12-2023 13-13-54

AFTER:
Oct-12-2023 12-09-28
Oct-12-2023 11-01-33

Fixes #71877, #60645, https://github.com/grafana/support-escalations/issues/7749

### CODE REVIEW

**Comment by nmarrs** [comment]
Great work on this! Really cool to see you diving into go code :) 

are we sure we want to backport this to 10.0.x? I'm pretty sure we could just backport to 10.1.x and not 10.0.x as this is more or less a bug vs a security issue 🤔 does that seem correct?

**Comment by drew08t** [comment]
> Great work on this! Really cool to see you diving into go code :)
> 
> are we sure we want to backport this to 10.0.x? I'm pretty sure we could just backport to 10.1.x and not 10.0.x as this is more or less a bug vs a security issue 🤔 does that seem correct?

10.1 should cover the impacted users if I'm not mistaken.

File: pkg/services/store/http.go [hunk 1]
```diff
@@ -260,7 +260,8 @@ func (s *standardStorageService) doCreateFolder(c *contextmodel.ReqContext) resp
 func (s *standardStorageService) list(c *contextmodel.ReqContext) response.Response {
 	params := web.Params(c.Req)
 	path := params["*"]
-	frame, err := s.List(c.Req.Context(), c.SignedInUser, path)
+	// maxFiles of 0 will result in default from wrapper
```
Comment (jennyfana) [inline pkg/services/store/http.go:263]:
```suggestion
	// maxFiles of 0 will result in default behaviour from wrapper
```

File: public/app/features/canvas/elements/icon.tsx [hunk 2]
```diff
@@ -115,6 +115,7 @@ export const iconItem: CanvasElementItem<IconConfig, IconData> = {
         editor: ResourceDimensionEditor,
         settings: {
           resourceType: 'icon',
+          maxFiles: 2000,
```
Comment (nmarrs) [inline public/app/features/canvas/elements/icon.tsx:118]: can we maybe export this somewhere as we are using two places?

**Review by nmarrs — APPROVED** [review]
code looks great / works great locally! I noticed though that we are missing passing the `2000` limit to the resource picker in the value mappings icon mapping UI though 🤔 

https://github.com/grafana/grafana/assets/22381771/22c483bf-9e2d-4696-98ad-83f26916f325

File: public/app/features/canvas/elements/icon.tsx (hunk 2 above)
Comment (drew08t) [reply public/app/features/canvas/elements/icon.tsx:118]: Good idea, was thinking the same! Do you have a suggestion on where this could live?

Comment (nmarrs) [reply public/app/features/canvas/elements/icon.tsx:118]: hm maybe as a const `FolderPickerTab` (the furthest down level where we are actually using this value for the time being?)

File: pkg/services/store/tree.go [hunk 3]
```diff
@@ -224,7 +224,7 @@ func (t *nestedTree) ListFolder(ctx context.Context, orgId int64, path string, a
 		)
 	}
 
-	listResponse, err := store.List(ctx, path, nil, &filestorage.ListOptions{
+	listResponse, err := store.List(ctx, path, &filestorage.Paging{First: maxFiles}, &filestorage.ListOptions{
```
Comment (ryantxu) [inline pkg/services/store/tree.go:227]: First?  this does not seem right

**Review by ryantxu — APPROVED** [review]
LGTM -- i was confused by the "First" parameter, and renamed it to "Limit"

**Pull request merged**